src/vs/workbench/contrib/terminal/browser/media/shellIntegration-rc.zsh

```diff
@@ -23,7 +23,7 @@ if [[ "$VSCODE_INJECTION" == "1" ]]; then
 		ZDOTDIR=$VSCODE_ZDOTDIR
 	fi
 
-	if [[ -f $USER_ZDOTDIR/.zsh_history ]]; then
+	if [[ -f $USER_ZDOTDIR/.zsh_history && -z HISTFILE ]]; then
```

Comment:
There's a missing `$` here:

```
if [[ -f $USER_ZDOTDIR/.zsh_history && -z $HISTFILE ]]; then
```

`-z HISTFILE` will always fail since it's checking the string `HISTFILE`, not the variable.

Reply:
Thanks for spotting @dmorgan-fa, we'll get this fix in for the release https://github.com/microsoft/vscode/issues/167575

Reply:
Thanks @dmorgan-fa for pointing the typo.